src/ImageArchive/Image.php

```diff
@@ -0,0 +1,240 @@
+			'caption'                       => $this->caption,
+			'credit'                        => $this->credit,
+			'restrictions'                  => $this->restrictions,
+			'sizes'                         => $this->sizes,
+			'wordpress_id'                  => $this->wordpress_id,
+			'original'                      => $this->original,
+			'original_language_title'       => $this->original_language_title,
+			'original_language_description' => $this->original_language_description,
+		];
+	}
+
+	/**
+	 * Accessor for title.
+	 *
+	 * @return mixed The title.
+	 */
+	public function get_title() {
+		return $this->title;
+	}
+
+	/**
+	 * Get the largest available size of this image and put it into WordPress (unless it's already in there).
+	 *
+	 * @param bool $use_original_language If true import with original language title and description, else in English.
+	 *
+	 * @throws UploadFailed When wp_upload_bits returns an error.
+	 * @throws RuntimeException When inserting attachment fails.
+	 */
+	public function put_in_wordpress( bool $use_original_language ): void {
+		if ( null !== $this->wordpress_id ) {
+			return;
+		}
+		$url = $this->original->getUrl();
+
+		$filename = basename( $url );
+		$filename = preg_replace( '/\?.*$/', '', $filename );
+
+		$context = stream_context_create();
+
+		// Upload file into WP upload dir.
+		$upload_file = wp_upload_bits( $filename, null, file_get_contents( $url, false, $context ) ); //phpcs:ignore WordPress.WP.AlternativeFunctions.file_get_contents_file_get_contents
+
+		if ( ! empty( $upload_file['error'] ) ) {
+			throw new UploadFailed( 'File upload failed: ' . $upload_file['error'] );
+		}
+
+		$wp_filetype = wp_check_filetype( $filename, null );
+
+		$title       = $use_original_language ? $this->original_language_title : $this->title;
+		$description = $use_original_language ? $this->original_language_description : $this->caption;
+
+		// Prepare an array of post data for the attachment.
+		$attachment = [
+			'post_mime_type' => $wp_filetype['type'],
+			'post_title'     => preg_replace( '/\.[^.]+$/', '', $title ),
+			'post_content'   => $description,
+			'post_excerpt'   => $description,
+			'post_status'    => 'inherit',
+		];
+
+		// Check title has full stop at the end, if not then add it.
+		$alt_text = rtrim( $title, '.' ) . '.';
+
+		$attachment_id = wp_insert_attachment( $attachment, $upload_file['file'], 0, true );
+
+		if ( is_wp_error( $attachment_id ) ) {
+			throw new RuntimeException( __( 'Error while inserting attachment...!', 'planet4-master-theme-backend' ) );
+		}
+
+		require_once ABSPATH . 'wp-admin/includes/image.php';
+
+		// Generate the metadata for the attachment, and update the database record.
+		$attachment_data = wp_generate_attachment_metadata( $attachment_id, $upload_file['file'] );
+
+		wp_update_attachment_metadata( $attachment_id, $attachment_data );
+
+		// Add credit to alt field.
+		if ( '' !== $this->credit ) {
```

Comment:
`!empty(trim())` in case api gives us something strange ? (or maybe do this in property assignment)